compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -438,23 +438,26 @@ DiagnosedSilenceableFailure
 transform_dialect::ForeachThreadToWorkgroupOp::applyToOne(
     func::FuncOp target, SmallVectorImpl<Operation *> &results,
     transform::TransformState &state) {
-  if (!isa<HAL::ExecutableOp, HAL::ExecutableVariantOp>(state.getTopLevel())) {
-    return mlir::emitDefiniteFailure(
-        state.getTopLevel(),
-        "requires HAL::ExecutableOp or HAL::ExecutableVariantOp toplevel "
-        "to attach the workgroup size information to a nested "
-        "ExecutableExportOp");
+  assert(target && "no target");
+  assert(state.getTopLevel() && "no top-level");
+  auto maybeExecutableVariant = getExecutableVariantOp(target);
+  if (failed(maybeExecutableVariant) ||
+      !state.getTopLevel()->isAncestor(*maybeExecutableVariant)) {
+    results.assign(1, target);
+    // Assign an empty handle and return success to allow propagating through
+    // functions that do not have an ExecutableVariantOp or an ExportOp (e.g.
+    // stream-related functions).
+    return DiagnosedSilenceableFailure::success();
```

Comment:
@ftynse, note that I turned these into a success rather than a silenceable failure to allow going through `func.func` ops that are not "mapped" i.e. the runtime-level `func.func`s with `stream` dialect ops, that call into the "mapped" `func.func` (those for which the ExecutableExportOp is retrievable).

This allows passing through but at some point, the `transform.structured.match` op gets multiple `func.func` targets, which is a harder stop: we probably don't want to starting thinking about cross-function matching and application.

Still investigating a bit before our chat but I think the solution is a mix of where should we apply the interpreter, where should we create the transform module, where should we apply the transform dialect (i.e. filter only to `func.func` that are mapped).